apps/public-docsite-v9/src/Concepts/Migration/KeepingDesignConsistent.stories.mdx

```diff
@@ -145,3 +145,43 @@ export const MyButton: ForwardRefComponent<ButtonProps> = React.forwardRef((prop
 ```
 
 This is the most advanced form of customization in Fluent UI React v9, but provides you complete control over components.
+
+## You can set custom style hooks per component type
+
+**⚠ Avoid custom style hooks**
+
+Prefer creating custom themes, setting className, replacing slots, extendeding theme tokens, and component recomposition.
+This approach is meant as an escape-hatch when all other customizations don't work.
+
+**⚠ Be careful **
+
+- Never change any state properties other than `className`.
+- Never dynamically change hooks at runtime. This will lead to unstable hook calls.
+- Custom style hooks are called from within the component and receive the component state.
+  Be judicious with creating styles and making `mergeClasses` call as you may affect the performance.
+
+You can pass `FluentProvider.customStyleHooks_unstable` a set of custom style hooks.
```

Comment:
might want to add that we still consider this API as unstable and can be changed/removed at any time